src/definition/populate.go

```diff
@@ -85,7 +85,14 @@ func IntegrationProtocol2PopulateFunc(
 					continue
 				}
 
-				ms := e.NewMetricSet(msType)
+				var ms metric.MetricSet
+				if len(e.Metrics) > 0 {
+					// This avoid generating a new ms for metrics collected from different sources like kubelet and ksm.
+					ms = e.Metrics[0]
```

Comment:
Oooh, so entities are shared globally. That's horrible and good at the same time :smile: This seems better than my merging code in the e2e tests (though mine has less production impact I guess).

Reply:
I agree , at the end i wanted to discuss this also with @roobre  since all these related changes will not be needed if moving to v3 happen , but not sure if we want that or not ( and also that has even more production impact)

Reply:
we agree on remove this approach and follow the one merging the metric sets in the e2e test.